infra/feast-operator/internal/controller/services/services.go

```diff
@@ -53,6 +53,11 @@ func (feast *FeastServices) Deploy() error {
 			if err := feast.deployFeastServiceByType(OfflineFeastType); err != nil {
 				return err
 			}
+			if _, shouldCreate := shouldCreatePvc(feast.FeatureStore, OfflineFeastType); !shouldCreate {
+				if err := feast.deleteOwnedFeastObj(feast.initPVC(OfflineFeastType)); err != nil && !apierrors.IsNotFound(err) {
+					return err
+				}
+			}
```

Comment:
let's move this block of code to the `deployFeastServiceByType` function instead... maybe as an `else {` to what you've already added?

Reply:
I initially was doing the same, but got worried about mixing a deletion during a deploy. If you are not worried about that, I can do as you suggest

Reply:
we're technically already mixing anyway... all good i think